gapic/src/main/com/google/gapid/MainWindow.java

```diff
@@ -267,6 +270,62 @@ private Point getDefaultInitialLocation(Point size) {
         Math.max(0, bounds.height - size.y) / 3);
   }
 
+  public void restartMainUi(Client client, Models models, Widgets widgets) {
+    Shell shell = getShell();
+
+    restartLoadingScreen(client, models, widgets);
+
+    showLoadingMessage("Restart UI...");
+    restartMenus(client, models, widgets);
+
+    mainUi = new LoadablePanel<MainViewContainer>(
+        mainArea, widgets, parent -> new MainViewContainer(parent, models, widgets));
+    models.capture.addListener(new Capture.Listener() {
+      @Override
+      public void onCaptureLoadingStart(boolean maintainState) {
+        shell.setText(Messages.WINDOW_TITLE + " - " + models.capture.getName());
+        setTopControl(mainUi);
+        mainUi.startLoading();
+      }
+
+      @Override
+      public void onCaptureLoaded(Message error) {
+        if (error != null) {
+          mainUi.showMessage(error);
+        } else {
+          // Let all other handlers of this event get a chance to process before we start disposing
+          // UI components underneath everybody.
+          scheduleIfNotDisposed(mainUi, () -> {
+            MainView view = mainUi.getContents().updateAndGet(
+                models.capture.getData().capture.getType());
+            view.updateViewMenu(findMenu(MenuItems.VIEW_ID));
+            getMenuBarManager().updateAll(true);
+            mainUi.stopLoading();
+          });
+        }
+      }
+    });
+
+    if (OS.isMac) {
+      MacApplication.init(shell.getDisplay(),
+          () -> showAbout(shell, models.analytics, widgets),
+          () -> showSettingsDialog(shell, models, widgets.theme),
+          file -> models.capture.loadCapture(new File(file)));
+    }
```

Comment:
This doesn't quite work correctly. `MacApplication.init` should only be called once. It adds listeners to the menu each time it is called.